src/libraries/System.Private.CoreLib/src/System/Numerics/IBinaryInteger.cs

```diff
@@ -37,7 +37,7 @@ static virtual TSelf LeadingZeroCount(TSelf value)
                 return TSelf.CreateChecked(bitCount);
             }
 
-            return (bitCount - TSelf.One) ^ TSelf.Log2(value);
+            return TSelf.IsNegative(value) ? TSelf.Zero : ((bitCount - TSelf.One) ^ TSelf.Log2(value));
```

Comment:
Log2 throws for negative integer values, so we need to explicitly check "is negative"